clang/include/clang/Basic/AttrDocs.td

```diff
@@ -455,6 +455,64 @@ The SYCL kernel in the previous code sample meets these expectations.
   }];
 }
 
+def SYCLKernelEntryPointDocs : Documentation {
+  let Category = DocCatFunction;
+  let Content = [{
+The ``sycl_kernel_entry_point`` attribute specifies that a function definition
+defines a pattern for an offload kernel entry point function to be emitted when
+the source code is compiled with ``-fsycl`` for a device target. Such functions
+serve as the execution entry point for a SYCL run-time library to invoke a SYCL
+kernel on a device. The function's parameters define the parameters to the
+offload kernel.
+
+The attribute requires a single type argument that specifies a class type that
+meets the requirements for a SYCL kernel name as described in section 5.2,
+"Naming of kernels", of the SYCL 2020 specification. A unique kernel name type
+is required for each function declared with the attribute. The attribute may
+not first appear on a declaration that follows a definition of the function.
+
+The attribute appertains only to non-member functions and static member
+functions that meet the following requirements:
+
+- Has a ``void`` return type.
+- Is not a variadic function.
+- Is not a coroutine.
+- Is not defined as deleted or as defaulted.
+- Is not declared with the ``constexpr`` or ``consteval`` specifiers.
+- Is not declared with the ``[[noreturn]]`` attribute.
+
+This attribute is intended for use in the implementation of SYCL run-time
```

Comment:
Is this something we should enforce?  That is, make this legal only in a `system-header` file?

Reply:
I don't think so. That would tend to bias against SYCL library implementations that aren't provided with LLVM/Clang or distributed with an OS. I don't know that such implementations are likely to appear, but I'm not inclined towards adding roadblocks for them.

Reply:
Erich, I'm going to consider this conversation about constraining use of the attribute to system headers resolved. If you have further thoughts on it, please unresolve.

Reply:
Again, no idea how I missed the response.  I think your answer here is sensible.